### PR TITLE
Fix deadlock in SystemEventsTests.CreateTimerTests

### PR DESCRIPTION
**Pull request by stephentoub**
The main thread in the ConcurrentTimers takes a lock on object timersSignaled and then calls CreateTimer while holding the lock.  CreateTimer tries to SendMessageW a message to be processed by the events window, and as a result won’t return until the window has processed the message.  Meanwhile, that window’s WindowThreadProc is handling a timer callback, which in the event handler in ConcurrentTimers tries to take the timersSignaled lock.  That lock is held by the main thread, that’s calling CreateTimer and doing the SendMessageW that will only wake up when this callback completes.  Deadlock.

The fix is to stop locking while performing such work (the locking was introduced a few months back in https://github.com/dotnet/corefx/pull/34790).

cc: @ViktorHofer, @maryamariyan, @ericstj

### CODE REVIEW

File: src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs [hunk 1]
```diff
@@ -78,24 +80,19 @@ public void TimerElapsedSignaled()
         public void ConcurrentTimers()
         {
             const int NumConcurrentTimers = 10;
-            var timersSignalled = new Dictionary<IntPtr, bool>();
+            var timersSignaled = new ConcurrentDictionary<IntPtr, bool>();
```
Comment (danmosemsft) [inline src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs]: This seemed wrong to my UK eyes, but it seems 'signaled' is the US spelling..

File: src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs [hunk 2]
```diff
@@ -104,33 +101,30 @@ public void ConcurrentTimers()
...
-                        SystemEvents.KillTimer(timer);
-                    }
+                    Assert.True(timersSignaled.TryGetValue(timer, out _));
+                    SystemEvents.KillTimer(timer);
                 }
             }
             finally
             {
                 SystemEvents.TimerElapsed -= handler;
-                elapsed.Dispose();
```
Comment (danmosemsft) [inline src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs]: Why not dispose anymore?

Comment (stephentoub) [reply src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs]: It's not strictly necessary (i.e. there aren't actually any resources to clean up in this case), and I was concerned about some subsequent handler coming along and trying to use it after it's been disposed, but after the dust here settled it's actually not an issue anymore, so I could put it back.

Comment (danmosemsft) [reply src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs]: I don't mind either way, it just catches the eye whenever there's something I know is disposable that the code makes no plan to dispose. Plus, the implementation of the disposable thing could in theory change, making disposing more important.

**Comment by danmosemsft** [comment]
@ViktorHofer 
```
D:\a\1\s\.dotnet\sdk\3.0.100-preview3-010431\NuGet.targets(119,5): error : Failed to retrieve information about 'transport.Microsoft.NETCore.Runtime.CoreCLR' from remote source 'https://dotnetfeed.blob.core.windows.net/dotnet-coreclr/flatcontainer/transport.microsoft.netcore.runtime.coreclr/index.json'. [D:\a\1\s\external\runtime\runtime.depproj]
D:\a\1\s\.dotnet\sdk\3.0.100-preview3-010431\NuGet.targets(119,5): error :   A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. [D:\a\1\s\external\runtime\runtime.depproj]

```

**Comment by danmosemsft** [comment]
cc @BruceForstall who is well acquainted with this hang.

**Comment by ViktorHofer** [comment]
This hang is expected at this point. We are working on a fix.

**Pull request merged**